runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -92,47 +92,67 @@ def create_template(self):
         lambda_function_associations = []
 
         if self.directory_index_specified:
-            rewrite_role = self.add_index_rewrite_role()
-            index_rewrite = self.add_cloudfront_directory_index_rewrite(rewrite_role)
-            index_rewrite_version = self.add_cloudfront_directory_index_rewrite_version(
-                index_rewrite
-            )
+            index_rewrite = self._get_index_rewrite_role_function_and_version()
             lambda_function_associations = self.get_directory_index_lambda_association(
                 lambda_function_associations,
-                index_rewrite_version
+                index_rewrite['version']
             )
 
         # Auth@Edge Lambdas
-        lambda_execution_role = self.add_lambda_execution_role()
         check_auth_lambda = self.get_auth_at_edge_lambda_and_ver(
             'CheckAuth',
             'Check Authorization information for request',
             'check_auth',
-            lambda_execution_role
+            self.add_lambda_execution_role(
+                'CheckAuthLambdaExecutionRole',
+                Join(':', ['arn:aws:logs:*',
+                           AccountId,
+                           'log-group:/aws/lambda/*-CheckAuth-*:*'])
```

Comment:
(thoughts here will apply to each of the functions)

1) Should probably include the [StackName](https://github.com/cloudtools/troposphere/blob/2.6.0/troposphere/__init__.py#L518) psuedoparam as a prefix here before the resource id (`CheckAuth` in this case) to keep it narrowed to its namespace
2) Would be good to explicitly link the function resource name (4 lines above, the `'CheckAuthLambdaExecutionRole'` string) to its use here (i.e. make it a variable and reference the variable)
3) For normal lambda function operation, it needs to start with `/aws/lambda/FUNCTIONAME` and then the lambda@edge executions need `/aws/lambda/REGION.FUNCTIONAME` -- to really go all out here (and we're already in for a penny ....) I think we'll want to provide the resource of `/aws/lambda/StackName-ResourceName-*` & `/aws/lambda/*.StackName-ResourceName-*`
4) Not a separate change from what was noted in point 3 ^, but just to highlight something implied by it: the CreateLogGroup & CreateLogStream permissions apply to log-group ARN (e.g. `/aws/lambda/StackName-FunctionResource-*`), and the PutLogEvents permissions should apply to the log-stream ARNs (`/aws/lambda/StackName-FunctionResource-*:log-stream:*`). So the ARN can't be `...log-group:/aws/lambda/*-CheckAuth-*:*` for all 3. Easiest option to just have a single resource ARN that will cover both is to omit the `:*` at the end